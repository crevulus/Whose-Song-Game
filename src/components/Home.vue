<template>
  <div class="flex mx-auto" style="max-width: 800px">
    <div class="w-full md:w-1/2 m-auto p-5">
      <ol class="list-decimal list-inside">
        <li>
          <strong>Call your colleagues!</strong> Use Hangouts, Skype, Zoom etc. and put the window to one side.
        </li>
        <li>
          <strong>Enter your name</strong> and create a new game room.
        </li>
        <li>
          <strong>Search for your favourite song</strong> and press confirm once you've chosen.
        </li>
        <li>
          <strong>Copy the link</strong> and share with your colleagues, then wait for them to join.
        </li>
        <li>
          When you're ready:
          <strong>hit start and play!</strong>
        </li>
      </ol>
      <h3 class="text-gray-800">Enter your name to start a game</h3>
      <div class="input-container">
        <t-input
          class="w-full mb-4 focus:border-black"
          v-model="name"
          maxlength="50"
          placeholder="Name..."
          @keyup.enter="createActivityInstance"
        />
        <t-button
          id="create-activity_home"
          class="w-full mb-8"
          variant="primary"
          @click="createActivityInstance"
          :disabled="name.length < 1"
        >Create activity room</t-button>
      </div>
    </div>
    <div class="hidden md:block">
      <iframe
        class="rounded-sm"
        src="https://open.spotify.com/embed/track/6sT9MWlJManry3EQwf4V80"
        width="300"
        height="380"
        frameborder="0"
        allowtransparency="true"
        allow="encrypted-media"
      ></iframe>
    </div>
  </div>
</template>
<script>
import { mapGetters } from "vuex";
import { API, graphqlOperation } from "aws-amplify";
import * as mutations from "@/graphql/mutations";
import variables from "@/variables";

export default {
  name: "Home",
  data() {
    return {
      name: ""
    };
  },
  computed: {
    ...mapGetters(["deviceId"])
  },
  methods: {
    async createActivityInstance() {
      const res = await API.graphql(
        graphqlOperation(mutations.createActivityInstance, {
          activityId: variables.activityId,
          hostId: this.deviceId,
          hostName: this.name
        })
      );
      const { activityInstanceId } = res.data.createActivityInstance;

      await API.graphql(
        graphqlOperation(mutations.whoseSongCreateActivityInstanceData, {
          activityInstanceId: activityInstanceId,
          userId: this.deviceId
        })
      );

      this.$router.push({
        name: "input",
        params: { activityInstanceId }
      });
    }
  }
};
</script>
