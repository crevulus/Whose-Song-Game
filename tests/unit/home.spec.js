import { mount, config, createLocalVue } from "@vue/test-utils";
import VueTailwind from "vue-tailwind";
import Home from "@/components/Home";

const localVue = createLocalVue();
localVue.use(VueTailwind);

config.mocks["$gtm"] = { trackView: function (name, path) { } };
config.mocks["$route"] = {};

describe("Home Component", () => {
  const wrapper = mount(Home, { localVue });

  it("renders Create button", () => {
    const msg = "Create";
    expect(wrapper.text()).toMatch(msg);
  });

  // snapshot testing
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
});
