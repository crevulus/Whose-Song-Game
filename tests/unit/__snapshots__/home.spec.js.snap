// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<div
  class="flex mx-auto"
  style="max-width: 800px;"
>
  <div
    class="w-full md:w-1/2 m-auto p-5"
  >
    <ol
      class="list-decimal list-inside"
    >
      <li>
        <strong>
          Call your colleagues!
        </strong>
         Use Hangouts, Skype, Zoom etc. and put the window to one side.
      
      </li>
       
      <li>
        <strong>
          Enter your name
        </strong>
         and create a new game room.
      
      </li>
       
      <li>
        <strong>
          Search for your favourite song
        </strong>
         and press confirm once you've chosen.
      
      </li>
       
      <li>
        <strong>
          Copy the link
        </strong>
         and share with your colleagues, then wait for them to join.
      
      </li>
       
      <li>
        
        When you're ready:
        
        <strong>
          hit start and play!
        </strong>
      </li>
    </ol>
     
    <h3
      class="text-gray-800"
    >
      Enter your name to start a game
    </h3>
     
    <div
      class="input-container"
    >
      <input
        class="w-full mb-4 focus:border-black t-input t-input-size-default t-input-status-default border block rounded p-3 bg-white"
        maxlength="50"
        placeholder="Name..."
        type="text"
      />
       
      <button
        class="w-full mb-8 t-button t-button-size-default border block rounded inline-flex items-center justify-center t-button-disabled cursor-not-allowed opacity-75 px-6 py-3 text-white bg-blue-500 border-blue-500 hover:bg-blue-600 hover:border-blue-600"
        disabled="disabled"
        id="create-activity_home"
        type="button"
      >
        Create activity room
      </button>
    </div>
  </div>
   
  <div
    class="hidden md:block"
  >
    <iframe
      allow="encrypted-media"
      allowtransparency="true"
      class="rounded-sm"
      frameborder="0"
      height="380"
      src="https://open.spotify.com/embed/track/6sT9MWlJManry3EQwf4V80"
      width="300"
    />
  </div>
</div>
`;
